edi/models/edi_exchange_consumer_mixin.py

```diff
@@ -0,0 +1,104 @@
+        domain=lambda r: [("model", "=", r._name)],
+    )
+    exchange_record_count = fields.Integer(compute="_compute_exchange_record_count")
+    missing_exchange_records = fields.Boolean(
+        compute="_compute_has_missing_exchange_records"
+    )
+
+    def _get_exchange_record_configuration(self):
+        """ This function should return a dict containing backend and exchange_type.
+        Example::
+        {
+            'configuration_identifier': {
+                'exchange_type': 'exchange_type_code', # required, string
+                'backend': backend # required, a `edi.backend` object,
+                'create_vals': {} # not required, dict with extra values for the
+                                  # creation vals
+            }
+        }
+        To be overridden by the modules.
+        We might need to check if the record has already been generated.
+        You can check an example on the test
+        """
+        self.ensure_one()
+        return {}
+
+    def _has_exchange_record(self, exchange_type, backend, extra_domain=False):
+        """This function is useful when generating the configuration"""
+        domain = [
+            ("model", "=", self._name),
+            ("res_id", "=", self.id),
+            ("backend_id", "=", backend.id),
+            ("type_id.code", "=", exchange_type),
+        ]
+        if extra_domain:
+            domain += extra_domain
+        return self.env["edi.exchange.record"].search(domain, limit=1)
+
+    @api.depends("exchange_record_ids")
+    def _compute_has_missing_exchange_records(self):
+        for record in self:
+            record.missing_exchange_records = bool(
+                record._get_exchange_record_configuration()
+            )
+
+    def _generate_exchange_records(self):
+        """This function will generate the exchange records of consumers.
+
+        For example, this can be added when posting an invoice or validating a picking
+        """
+        exchange_records = self.env["edi.exchange.record"]
+        for record in self:
+            configurations = record._get_exchange_record_configuration()
+            for config_id in configurations:
+                configuration = configurations[config_id]
+                exchange_type = configuration["exchange_type"]
+                backend = configuration["backend"]
+                creation_args = {
+                    "model": record._name,
+                    "res_id": record.id,
+                }
+                creation_args.update(configuration.get("create_vals", {}))
+                exchange_record = backend.create_record(exchange_type, creation_args)
+                record._post_generate_exchange_records(exchange_record)
+                exchange_records |= exchange_record
+        return exchange_records
+
+    def _post_generate_exchange_records(self, exchange_record):
+        if (
+            self.env.context.get("force_edi_exchange_generation", False)
```

Comment:
I am not sure if this context should be taken from the record or from the exchange_record (would be inherited from the backend of the configuration)